corehq/apps/es/users.py

```diff
@@ -118,3 +119,14 @@ def primary_location(location_id):
             filters.term('domain_memberships.location_id', location_id)
         ),
     )
+
+
+def location(location_id):
+    # by any assigned-location primary or not
+    return filters.OR(
+        filters.AND(mobile_users(), filters.term('assigned_location_ids', location_id)),
+        filters.AND(
+            web_users(),
+            filters.term('domain_memberships.assigned_location_ids', location_id)
+        ),
+    )
```

Comment:
Did you confirm that this works as expected when the passed in `location_id` is a list of location ids?  That is, will
```python
query.location(['123', '456'])
```
Match a user where
```python
location_ids = ['abc', '123']
```
I'm not sure how the `terms` filter compares two lists.